pkg/actuators/machine/utils.go

```diff
@@ -153,28 +156,42 @@ func correctExistingTags(machine *machinev1.Machine, instance *ec2.Instance, cli
 			if *tag.Key == "kubernetes.io/cluster/"+clusterID && *tag.Value == "owned" {
 				clusterTagOk = true
 			}
+			if _, present := tags[*tag.Key]; present {
```

Comment:
I believe if a tag key exists but it's value has change due to this logic it won't be updated. Also please add test cases for when tags are added and updated.

Reply:
Done